performance/wazuh_manager_mockers/src/manager_mock/simulate_manager.py

```diff
@@ -0,0 +1,280 @@
+
+    command = [
+        sys.executable,
+        f"{service_path}.py",
+        "--port", str(port),
+        "--database-path",
+        database_path,
+        "--key",
+        key_path,
+        "--cert",
+        cert_path,
+    ]
+    if debug:
+        command.extend(['-v'])
+
+    logging.info(f"Starting {service_name} on port {port}...")
+    return subprocess.Popen(command)
+
+
+def run_agent_comm(port: str, database_path: str, certs_path: str, report_path: str,
+                   api_version: str, debug: bool = False) -> subprocess.Popen:
+    """Starts a mock agent communication service as a subprocess.
+
+    Args:
+        port (int): The port number on which the agent communication service will listen.
+        database_path (str): Path to the database file.
+        certs_path (str): Path to the directory containing SSL certificates.
+        report_path (str): Path to the CSV file where metrics will be reported.
+        api_version (str): agent comm version
+        debug (bool): Enable debug.
+
+    Returns:
+        subprocess.Popen: The subprocess object representing the started agent communication process.
+
+    Raises:
+        FileNotFoundError: If the script or certificates are not found.
+    """
+    service_name = 'agent_comm_mock'
+    service_path = Path(__file__).resolve().parent.parent / 'manager_mock_services' / service_name / service_name
+    cert_path = os.path.join(certs_path, 'cert.pem')
+    key_path = os.path.join(certs_path, 'private_key.pem')
+
+    command = [
+        sys.executable,
+        f"{service_path}.py",
+        "--port", str(port),
+        "--database-path",
+        database_path,
+        "--key",
+        key_path,
+        "--cert",
+        cert_path,
+        '--report-path',
+        report_path,
+        '--api-version',
+        api_version
+    ]
+    if debug:
+        command.extend(['-v'])
+    print(command)
+
+    logger.info(f"Starting {service_name} on port {port}...")
+    return subprocess.Popen(command)
+
+
+def generate_certificates(server_path: str) -> str:
+    """Creates and returns the path to SSL certificates for server communication.
+
+    Args:
+        server_path (str): Path where certificates will be created. If not provided, a temporary
+        directory will be created.
+
+    Returns:
+        str: Path to the directory containing the generated SSL certificates.
+
+    Raises:
+        OSError: If there is an error creating directories or files.
+    """
+    if not server_path:
+        logger.info("Creating server directory")
+        server_path = tempfile.mkdtemp()
+        logger.info(server_path)
+
+    credentials_path = os.path.join(server_path, 'certs')
+    if os.path.exists(credentials_path):
+        logger.info("Detected existing certs. Removing")
+        shutil.rmtree(credentials_path)
+
+    os.mkdir(credentials_path)
+    create_manager_credentials(credentials_path)
+
+    return credentials_path
+
+
+def main():
```

Comment:
Type hints in tests or main functions are not useful. I will suggest to avoid use type hinting in these special cases in https://github.com/wazuh/wazuh-qa-automation/issues/42